prover-server/src/prove.rs

```diff
@@ -0,0 +1,141 @@
+use crate::spec::ProofType;
+use crate::utils::{kroma_info, kroma_msg};
+use jsonrpc_core::Result;
+use rand_core::SeedableRng;
+use rand_xorshift::XorShiftRng;
+use serde::{Deserialize, Serialize};
+use std::fs;
+use std::fs::create_dir_all;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use types::eth::BlockTrace;
+use utils::Measurer;
+use zkevm::circuit::{EvmCircuit, StateCircuit, SuperCircuit, AGG_DEGREE, DEGREE};
+use zkevm::io::write_file;
+use zkevm::prover::{AggCircuitProof, Prover, TargetCircuitProof};
+use zkevm::utils::{load_or_create_params, load_or_create_seed};
+
+const PARMA_DIR: &str = "./test_params/";
+const SEED_FILE: &str = "./test_seed";
+const OUT_PROOF_DIR: &str = "./out_proof/";
+const VERIFIER_NAME: &str = "zk-verifier.sol";
+
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct ProofResult {
+    pub final_pair: Option<Vec<u8>>,
+    pub proof: Vec<u8>,
+}
+
+impl ProofResult {
+    pub fn new(proof: Vec<u8>, final_pair: Option<Vec<u8>>) -> Self {
+        Self { proof, final_pair }
+    }
+}
+
+pub fn create_proof(trace: BlockTrace, proof_type: ProofType) -> Result<ProofResult> {
+    // load or create material for prover
+    let params = load_or_create_params(PARMA_DIR, *DEGREE)
+        .unwrap_or_else(|_| panic!("{}", kroma_msg("fail to load or create params")));
+    let agg_params = load_or_create_params(PARMA_DIR, *AGG_DEGREE)
+        .unwrap_or_else(|_| panic!("{}", kroma_msg("fail to load or create params")));
+    let seed = load_or_create_seed(SEED_FILE)
+        .unwrap_or_else(|_| panic!("{}", kroma_msg("fail to load or create seed")));
+    let rng = XorShiftRng::from_seed(seed);
+
+    // prepare directory to store proof. (i.e., ./out_proof/<block_number>/)
+    let height_hex = trace.header.number.unwrap().to_string();
+    let out_dir = PathBuf::from(OUT_PROOF_DIR).join(height_hex);
+    let _ = create_dir_all(&out_dir);
+
+    // build prover
+    let mut prover = Prover::from_params_and_rng(params, agg_params, rng);
+    // specify the dir to store the vk and proof of the intermediate circuit.
+    prover.debug_dir = out_dir.to_str().unwrap().to_string();
+
+    match proof_type {
+        ProofType::None => {
+            panic!("Invalid proof type");
```

Comment:
```suggestion
            panic!("invalid proof type");
```
Above and below, you are using lowercases for panicking.